proof/invariant-abstract/AARCH64/ArchInvariants_AI.thy

```diff
@@ -535,6 +542,9 @@ definition hyp_live :: "kernel_object \<Rightarrow> bool" where
    | ArchObj ao \<Rightarrow> arch_live ao
    |  _ \<Rightarrow> False"
 
+definition arch_tcb_live :: "arch_tcb \<Rightarrow> bool" where
+  "arch_tcb_live arch_tcb \<equiv> tcb_cur_fpu arch_tcb"
```

Comment:
I think some kind of comment would be useful in this file that indicates the magic link between arch_tcb_live, is_tcb_cur_fpu and valid_cur_fpu, indicating whether a thread is the current FPU owner via ghost state and also indicating which thread is the FPU owner via global state

Reply:
I've added a couple of comments, let me know if they help or if you think it needs more. There's also the comment in the spec that is relevant and that people might come across if they're curious about the ghost variable.
https://github.com/seL4/l4v/blob/2cffea962d05fdbb99f9dfea5a27c73512c3bbbd/spec/abstract/AARCH64/Arch_Structs_A.thy#L370-L372

Reply:
Done. Do you think the comment on `valid_cur_fpu` is helpful or is it repeating the definition and not adding much?

Reply:
Cool, thanks.
It does a little, but it documents the intent. Looking at the definition and seeing it matches the intent gives me good feelings about it, so I'd leave it as is.